lib/client.js

```diff
@@ -915,7 +842,11 @@ MqttClient.prototype._handleConnack = function (packet) {
 
   if (rc === 0) {
     this.reconnecting = false
-    this.emit('connect', packet)
+    var that = this
+    this.connectEmitter = function () {
+      that.emit('connect', packet)
+    }
+    this._onConnect()
```

Comment:
You can pass the packet to `_onConnect(packet)`  and wrap it in the closures there.

Reply:
Thanks, I will commit it.